packages/website/public/llms/styleGuide.md

```diff
@@ -0,0 +1,417 @@
+
+- When you want to store complex data as code, use lists of dictionaries, instead of using lists of lists. This makes things clearer. For example, use:
+
+```squiggle
+[
+  {year: 2023, value: 1},
+  {year: 2024, value: 2},
+]
+instead of:
+[
+  [2023, 1],
+  [2024, 2],
+]
+```
+
+You can use lists instead when you have a very long list of items (20+), very few keys, and/or are generating data using functions.
+
+- Tables are a great way to display structured data.
+- You can use the '@showAs' tag to display a table if the table can show all the data. If this takes a lot of formatting work, you can move that to a helper function. Note that helper functions must be placed before the '@showAs' tag.
+
+For example:
+
+```squiggle
+@hide
+strategiesTable(data) = Table.make(
+  data,
+  {
+    columns: [
+      { name: "name", fn: {|f| f.n} },
+      { name: "costs", fn: {|f| f.c} },
+      { name: "benefits", fn: {|f| f.b} },
+    ],
+  }
+)
+
+@name("AI Safety Strategies")
+@doc("List of 10 AI safety strategies with their costs and benefits")
+@showAs(strategiesTable)
+strategies = [
+  { n: "AI Ethics", c: 1M to 5M, b: 5M to 20M },
+  { n: "Alignment Research", c: 2M to 10M, b: 10M to 50M },
+  { n: "Governance", c: 500k to 3M, b: 2M to 15M },
+  ...
+]
+```
+
+## Tags and Annotations
+
+### @name, @doc, @hide, @showAs
+
+- Use `@name` for simple descriptions and shortened units. Use `@doc` for further details (especially for detailing types, units, and key assumptions), when necessary. It's fine to use both @name and @doc on the same variable - but if so, don't repeat the name in the doc; instead use the doc() for additional information only.
+- In `@name`, add units wherever it might be confusing, like "@name("Ball Speed (m/s)"). If the units are complex or still not obvious, add more detail in the @doc().
+- For complex and important functions, use `@name` to name the function, and `@doc` to describe the arguments and return values. @doc should represent a docstring for the function. For example:
+
+```
+@doc("Adds a number and a distribution.
+\`\`\`squiggle
+add(number, distribution) -> distribution
+\`\`\`")
+```
+
+- Variables that are small function helpers, and that won't be interesting or useful to view the output of, should get a `@hide` tag. Key inputs and outputs should not have this tag.
+- Use `@showAs` to format large lists, as tables and to show plots for dists and functions where appropriate.
+
+### `@format()`
+
+- Use `@format()` for numbers, distributions, and dates that could use obvious formatting.
+- The `@format()` tag is not usable with dictionaries, functions, or lists. It is usable with variable assignments. Examples:
+
+```squiggle
+net_benefit(costs, benefits) = benefits - costs // not valid for @format()
+net_benefit = benefits - costs // valid for @format()
+```
+
+- This mainly makes sense for dollar amounts, percentages, and dates. ".0%" is a decent format for percentages, and "$,.0f" can be used for dollars.
```

Comment:
Maybe mention explicitly that `@format` uses `d3-format` specs?

Reply:
This is already in the docs, and so far, the LLM has always seemed to understand this.